public/app/features/dashboard/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { PureComponent, createRef } from 'react';
+import moment from 'moment';
+import {
+  TimeRange,
+  TimeOptions,
+  TimeOption,
+  SelectOptionItem,
+  ClickOutsideWrapper,
+  SelectButton,
+  HeadlessSelect,
+} from '@grafana/ui';
+import Popper from '@grafana/ui/src/components/Tooltip/Popper';
+
+import { TimePickerOptionGroup } from './TimePickerOptionGroup';
+import { TimePickerPopOver } from './TimePickerPopOver';
+import { mapTimeOptionToTimeRange, mapTimeRangeToRangeString } from './time';
+
+export interface Props {
+  value: TimeRange;
+  isTimezoneUtc: boolean;
+  popOverTimeOptions: TimeOptions;
+  selectTimeOptions: TimeOption[];
+  timezone?: string;
+  onChange: (timeRange: TimeRange) => void;
+  onMoveBackward: () => void;
+  onMoveForward: () => void;
+  onZoom: () => void;
+}
+
+export interface State {
+  isPopOverOpen: boolean;
+  isSelectOpen: boolean;
+  isSmallScreen: boolean;
+}
+
+export class TimePicker extends PureComponent<Props, State> {
+  pickerTriggerRef = createRef<HTMLDivElement>();
+  state = { isSelectOpen: false, isPopOverOpen: false, isSmallScreen: false };
```

Comment:
What do you think about introducing `TimePickerSelect` component that would hide some of the state that's manipulated here? Would Love to see TimePicker to be as dummy as possible with UI logic delegated to lower level pieces.

For controlling TimePickerPopover you can try using `PopperController` that hides the popovers showing/hiding logic.

Reply:
I've created a ButtonSelect component that is used in both TimePicker and RefreshPicker